docs/man/rpm-scriptlets.7.scd

```diff
@@ -0,0 +1,360 @@
+this can be overridden (see *OPTIONS*). Typically a scriptlet has a body for
+the interpreter to execute, but the interpreter can also be a standalone,
+special-purpose program, in which case the scriptlet body may be omitted.
+The scriptlet interpreter dependency is automatically added by RPM, but
+other dependencies to tools used by scriptlets must be manually added
+in the spec.
+
+Scriptlets cannot be interactive, and should not rely on their output
+to be seen by users.
+
+# ARGUMENTS
+_NAME_
+	The name of the (sub)package the scriptlet is associated with, within
+	the *rpm-spec*(5) file. If omitted, the scriptlet is associated with
+	the main package.
+
+	By default, subpackages are named by prepending the main package
+	name, followed by a dash, to the subpackage name. For example,
+	if the main package is called *myproject*, *myproject-*_NAME_.
+	This behavior can be changed with the *-n* option.
+
+_TRIGGERCOND_
+	A trigger condition is a name of a package, optionally followed
+	by a version range: _NAME_ [_op_ _EVR_] where _op_ is one of
+	*<*, *<=*, *=*, *>=* or *>* and _EVR_ is an *rpm-version*(7) label.
+	For example, *glibc < 2.40*.
+
+_PATHPREFIX_
+	A path prefix on which the file trigger is triggered on. The prefix
+	must be a complete path of its own, typically a directory whose
+	contents the trigger is interested in, eg. _/usr/lib_.
+
+# OPTIONS
+*-e*
+	Perform *rpm-macros*(7) expansion on the scriptlet before executing it.
+
+*-n*
+	_NAME_ is interpreted as the full package name.
+
+*-p* _EXECUTABLE_
+	Execute the scriptlet using _EXECUTABLE_ as the interpreter.
+
+	As a special case, specifying *<lua>* as the _EXECUTABLE_ invokes the
+	embedded Lua interpreter. The various special behaviors of such
+	scriplets are documented in the *rpm-lua*(7) manual.
+
+*-q*
+	Perform *rpm-queryformat*(7) on the package header, using the
+	scriptlet as the format.
+
+# FILETRIGGER OPTIONS
+*-P* _NUMBER_
+	Priority of the trigger, among the other file triggers of the same
+	kind. The default priority of file triggers is 100000. A larger
+	priority means earlier execution.
+
+# SCRIPTS
+Scripts execute unconditionally at their pre-determined slots in a transaction.
+
+Scripts receive one argument (*$*), which contains the number of installed
+instances of the package when the operation has completed.
+This can be used to detect new installations from upgrades inside the scripts.
+
+The scripts of a package can be examined with a *rpm -q --scripts* query.
+
+## Package scripts
+Package scripts execute before and after the main operation
+(install/remove etc) of the package inside a transaction.
+
+Package scripts should should only be used for actions that are fundamentally
+package specific. Domain specific registries, databases, caches and such are
+much better handled centrally by file triggers. Similarly, user and group
+creation should be handled by *rpm-sysusers*(7), rather than scripting them.
+
+*%pre*
```

Comment:
I wonder if we should have the full "signature" with each scriptlet here, like in SYNOPSIS, or just keep it in SYNOPSIS? To somebody who arrives at a specific scriptlet section, it may not be immediately obvious that the scriptlets actually take some arguments.

What I mean is, having e.g. the following here:

```
*%pre* [options] [_NAME_]
[body]
```

That said, doing this would make the scriptlet enumeration much more noisy, so I'm not sure it's worth it. Alternatively, we could just have a short sentence in each of scriptlet section that mentions that arguments can be passed to scriptlets and that points to the SYNOPSIS for details.

Reply:
Yeah I don't think we want to repeat it all for every scriptlet, but it's a fair point. We could maybe repeat the synopsis part in each of these sections so it's easy to check up.

Reply:
Ack, works for me :smile: